File: #728_selfDividingNumbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []
        for i in range(left, right+1):
            if '0' in str(i):
                continue
            divided = 0
            for j in str(i):
                if i%int(j) != 0:
                    divided = 1
            if divided == 0:
                res.append(i)
        return res

File: Java/#728_selfDividingNumbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> res = new ArrayList<Integer>();
        for(int i=left;i<right+1;i++){
                if(selfDividing(i)){
                    res.add(i);
                }
        }
        return res;
    }
    
    public boolean selfDividing(int n){
        int original=n;
        while(n>0){
            if(n%10==0) return false;
            if(original%(n%10)!=0) return false;
            n/=10;
            
        }
        return true;
    }
}
